pkg/metrics/metrics.go

```diff
@@ -0,0 +1,174 @@
+package metrics
+
+import (
+	"encoding/json"
+	"strings"
+
+	corelisters "k8s.io/client-go/listers/core/v1"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	configv1 "github.com/openshift/cluster-samples-operator/pkg/apis/samples/v1"
+	"github.com/openshift/cluster-samples-operator/pkg/client"
+	sampoplisters "github.com/openshift/cluster-samples-operator/pkg/generated/listers/samples/v1"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// per https://prometheus.io/docs/practices/naming/#metric-names feel our metrics here fall under the _info category of classification
+
+	// since we want to index by imagestream name, where each imagestream is either failing or not failing, this is why
+	// we went with the _info suffix vs. the _total suffix
+	failedImportsQuery = "openshift_samples_failed_imagestream_import_info"
+
+	degradedQuery = "openshift_samples_degraded_info"
+
+	invalidConfigQuery = "openshift_samples_invalidconfig_info"
+
+	invalidSecretQuery = "openshift_samples_invalidsecret_info"
+
+	missingSecret        = "missing-secret"
+	missingTBRCredential = "missing-tbr-credential"
```

Comment:
are we being inconsistent in use of underscore vs dash  here?

Reply:
I thought I saw the use of dash somewhere, but I can't find that now, and I just revisited https://prometheus.io/docs/practices/naming/ and their labels example has underscores as well

I'll change it

Reply:
dash to underscore conversion pushed @bparees